lib/src/components/NxTreeView/NxTreeView.tsx

```diff
@@ -4,55 +4,81 @@
  * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
  */
-import React, { FunctionComponent, HTMLAttributes } from 'react';
+import React, { FunctionComponent, useMemo, forwardRef } from 'react';
 import classnames from 'classnames';
+import { faCaretRight } from '@fortawesome/free-solid-svg-icons';
 
 import NxTooltip from '../NxTooltip/NxTooltip';
-import { Props, propTypes, childPropTypes } from './types';
-export { Props } from './types';
+import { getRandomId } from '../../util/idUtil';
+import { Props, NxTreeViewChildProps, propTypes, childPropTypes } from './types';
+import NxFontAwesomeIcon from '../NxFontAwesomeIcon/NxFontAwesomeIcon';
+
+export { Props, NxTreeViewChildProps } from './types';
+
 import './NxTreeView.scss';
 
 const NxTreeView: FunctionComponent<Props> =
   function NxTreeView(props) {
     const { onToggleCollapse, isOpen, disabled, children, triggerContent, triggerTooltip, className, id } = props;
 
-    const treeViewClasses = classnames('nx-tree-view', className, {
-          'nx-tree-view--expanded': isOpen,
-          'nx-tree-view--collapsed': !isOpen,
+    const isEmpty = !React.Children.count(children),
+        isExpanded = isOpen && !isEmpty, // conceptually we don't allow empty tree views to expand
+        treeViewClasses = classnames('nx-tree-view', className, {
+          'nx-tree-view--expanded': isExpanded,
+          'nx-tree-view--collapsed': !isExpanded,
           'nx-tree-view--disabled': disabled,
-          'nx-tree-view--empty': !React.Children.count(children)
+          'nx-tree-view--empty': isEmpty
         }),
+        treeViewId = useMemo(() => id || getRandomId('nx-tree-view'), [id]),
         trigger = (
-          <div className="nx-tree-view__trigger" onClick={onToggleCollapse || undefined}>
-            <div className="nx-tree-view__twisty">
-              <span className="nx-tree-view__twisty-icon"/>
-            </div>
+          <button className="nx-tree-view__trigger"
+                  onClick={onToggleCollapse || undefined}
+                  aria-controls={treeViewId}
+                  aria-expanded={isExpanded}
+                  aria-disabled={disabled || isEmpty || undefined}>
+            <NxFontAwesomeIcon className="nx-tree-view__twisty" icon={faCaretRight} />
             <div className="nx-tree-view__text">
               {triggerContent}
             </div>
-          </div>
+          </button>
         ),
         triggerTooltipProps = typeof triggerTooltip === 'string' ? { title: triggerTooltip } : triggerTooltip;
 
     return (
-      <div className={treeViewClasses}
-           id={id || undefined}>
+      <div className={treeViewClasses} id={treeViewId} role="tree">
         { triggerTooltipProps ? <NxTooltip { ...triggerTooltipProps } >{trigger}</NxTooltip> : trigger }
-        <div className="nx-tree-view__children">
+        <div className="nx-tree-view__children" role="group">
```

Comment:
In the W3 example `role="tree item"` appeared at this level. I experimented with this locally and found that it made no difference at all as far as Voice Over was concerned, and frankly I find that how you have it makes more sense semantically how you have it.

Reply:
> In the W3 example role="tree item" appeared at this level. I experimented with this locally and found that it made no difference at all as far as Voice Over was concerned, and frankly I find that how you have it makes more sense semantically how you have it.

Keep in mind that the concept they have of what a tree is is a little bit different than our concept.  I felt like putting this role here made sense given how we have it set up overall.